### PR TITLE
Adding pkce support field

### PR DESCRIPTION
**Pull request by uhunnyslack**
###  Summary
To support more connectors, 3p auth manifest config requires some more new SDK fields

This PR is to add a new field that helps the connector team unblock creating some more connectors.

`use_pkce` : boolean -> this is true for providers that uses PKCE:

More discussions can be found in our slack instance.

### Important
Please do not merge it until the web app changes are merged! Thanks!

### Requirements (place an `x` in each `[ ]`)

* [X] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/deno-slack-sdk/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: src/manifest/manifest_test.ts
Comment (filmaj) [inline src/manifest/manifest_test.ts:1012]: I think the only thing this test tests is that the provider key is set properly on the exported manifest. Other than that assertion, this test will always pass, as the manifest values for any oauth provider will always be the output of the provider's `export` method. For example, in the future, if I change the [`Provider.export` method](https://github.com/slackapi/deno-slack-sdk/blob/main/src/providers/oauth2/mod.ts#L28) to `delete` the `pkce` field, this test will still pass.

I think we want to change this test to ensure that the `options.use_pkce` field is set properly on the manifest in both cases: it is `true` in one case and `false` in the other.

**Review by filmaj — CHANGES_REQUESTED** [review]
I think the test needs changing as it will always pass, but other than looks fine to me.

File: src/manifest/manifest_test.ts
Comment (uhunnyslack) [reply src/manifest/manifest_test.ts:1012]: I tried with some test mutation and seems like I can make the test fail like this:
Screenshot 2023-08-23 at 1 35 15 PM

Comment (uhunnyslack) [reply src/manifest/manifest_test.ts:1012]: Do you think this would work? ☝️

Comment (filmaj) [reply src/manifest/manifest_test.ts:1012]: Right, that fails because the key of Provider2 is being checked against the export of Provider1, so we are checking that two different things are equal.

Comment (filmaj) [reply src/manifest/manifest_test.ts:1012]: I'd suggest enumerating specifically the fields that you expect in each case. Something like:

```
assertEquals(exportedManifest.external_auth_providers.oauth2.test_provider_with_with_pkce_true.options.use_pkce, true);
assertEquals(exportedManifest.external_auth_providers.oauth2.test_provider_with_with_pkce_false.options.use_pkce, false);
```

Comment (uhunnyslack) [reply src/manifest/manifest_test.ts:1026]: Thanks for the suggestion! Added the changes ! 🙇‍♀️

Comment (uhunnyslack) [reply src/manifest/manifest_test.ts:1026]: @filmaj  Please do not merge it until the web app changes are merged! Thanks!

**Review by filmaj — APPROVED** [review]

**Comment by filmaj** [comment]
@uhunnyslack I've assigned this PR to you and added the do not merge label to it. Feel free to merge whenever you are ready.

**Pull request merged**
